mlir/lib/Conversion/GENToSPIRV/GENToSPIRV.cpp

```diff
@@ -57,10 +58,19 @@ class GEN3DNDRangeLoweringBase : public ConversionPattern {
         this->template getTypeConverter<SPIRVTypeConverter>()->getIndexType();
     constexpr StringLiteral spvBuiltinPrefix = "__spirv_BuiltIn";
     constexpr StringLiteral spvBuiltinSuffix = "";
+
     Value vector = spirv::getBuiltinVariableValue(
         op, builtin, builtinType, rewriter, spvBuiltinPrefix, spvBuiltinSuffix);
-    rewriter.replaceOpWithNewOp<spirv::VectorExtractDynamicOp>(op, vector,
-                                                               operands[0]);
+    Value idx = operands[0];
+
+    if (IntegerAttr idxAttr; matchPattern(idx, m_Constant(&idxAttr))) {
+      // Don't need to check the range here because that is verified already
+      rewriter.replaceOpWithNewOp<spirv::CompositeExtractOp>(
+          op, vector, idxAttr.getValue().getZExtValue());
```

Comment:
```suggestion
    if (llvm::APInt idx; matchPattern(idx, m_Constant(&idx))) {
      // Don't need to check the range here because that is verified already
      rewriter.replaceOpWithNewOp<spirv::CompositeExtractOp>(
          op, vector, idx.getZExtValue());
```

Reply:
Please add an `assert` to double check the value. Upstream PR got comments on that and I reverted (not pulled downstream tho). We would just play it safe in case that changes in this dialect in the future.